src/serlo-editor/math/editor.tsx

```diff
@@ -74,8 +77,9 @@ export interface MathEditorProps {
   onEditorChange(visual: boolean): void
   onInlineChange?(inline: boolean): void
   onChange(state: string): void
-  onMoveOutRight?(): void
-  onMoveOutLeft?(): void
+
+  onMoveOutRight: (options?: { closeThroughModal?: boolean }) => void
```

Comment:
I'd prefer if you would define a new handler, like `onMathEditorClose` or similar, and then use that new one for the Esc hotkey and close button. Reasons:
1. `onMoveOutRight` is a very specific handler for MathQuill package
2. A new handler could have a descriptive name that explains what it handles, instead of the vague `onMoveOutRight`

Will continue this comment in another place.